osclib/cache.py

```diff
@@ -273,7 +285,7 @@ def path(url, project, include_file=False, makedirs=False):
             os.makedirs(directory)
 
         if include_file:
-            parts.append(hashlib.sha1(url).hexdigest())
+            parts.append(hashlib.sha1(url.encode('utf-8')).hexdigest())
```

Comment:
Something is wrong with caching here, if I create caching without this PR and then switch to this is utilizes it properly, but if I wipe the request cache and then allow it to re-create it fails to find cache entries repeatedly.